web_external/views/body/SubmitView.js

```diff
@@ -184,16 +206,16 @@ var SubmitView = View.extend({
     },
 
     uploadFinished: function () {
-        var submission = new SubmissionModel();
-        submission.on('c:submissionPosted', function () {
-            router.navigate(`submission/${submission.id}`, {trigger: true});
+        this.submission.on('c:submissionPosted', function () {
+            router.navigate(`submission/${this.submission.id}`, {trigger: true});
         }, this).postSubmission({
             phaseId: this.phase.id,
             folderId: this.folder.id,
             title: this.title,
             organization: this.organization,
             organizationUrl: this.organizationUrl,
-            documentationUrl: this.documentationUrl
+            documentationUrl: this.documentationUrl,
+            approach: this.approach
```

Comment:
There is now some unexpected behavior when submitting:
- If I leave the approach field blank, the empty string is submitted, but the server only treats a null approach name as the "default" approach. The empty string should also indicate the "default" approach.
- The text field allows entering any text, but upon submission the approach is transformed to lowercase. It would be nice if the web client enforced this in some way, or if the server didn't transform the case. Otherwise this will lead to odd results, for example a "H-DenseUNet" approach would display as "h-denseunet".

Reply:
For the first point, I could fix the issue here, but I think you are right... I should change the endpoint to recognize the empty string in the same was as null.

For the second point, I somewhat arbitrarily chose to make the approach coerce to lower case.  Trying to keep the approach name case insensitive but maintain the original representation leads to a lot of tricky complications that I think are worth dealing with.  To resolve the confusion, I'll probably just make the approach name case sensitive in the database.